libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -49,6 +49,13 @@ string ABIFunctions::tupleEncoder(
 	options.padded = true;
 	options.dynamicInplace = false;
 
+	for (Type const*& t: _targetTypes)
+	{
+		solAssert(t, "");
+		t = t->fullEncodingType(options.encodeAsLibraryTypes, true, !options.padded);
```

Comment:
This was only done in `encodingFunction` further down the call chain...

Reply:
Because of this, we queried the check https://github.com/ethereum/solidity/pull/10733/files#diff-c10bb1823788cdc6974420487177f1a6badb1a26da48fb36f0ad49d37422000dR88 on the wrong type.